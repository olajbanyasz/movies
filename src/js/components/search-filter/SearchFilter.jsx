import React, { Component } from 'react';
import { ButtonToolbar, ToggleButtonGroup, ToggleButton } from 'react-bootstrap';
import './searchfilter.style.less';

class SearchFilter extends Component {

  state = {
    searchby: null
  }

  changeHandler = (searchby) => {
    if (searchby !== this.state.searchby) {
      this.setState({searchby});
      this.props.changeHandler(searchby);
    }
  };

  render () {
    return (
      <div className="search-filter">
        <span>SEARCH BY</span>
        <ButtonToolbar className='search-filter-buttons'>
          <ToggleButtonGroup
            type="radio"
            name='searchfilter'
            defaultValue={this.props.searchby}
            onChange={this.changeHandler}
          >
            <ToggleButton value={'TITLE'} className='search-by-title'>TITLE</ToggleButton>
            <ToggleButton value={'GENRE'} className='search-by-genre'>GENRE</ToggleButton>
          </ToggleButtonGroup>
        </ButtonToolbar>
      </div> 
    );
  }
}

export default SearchFilter;