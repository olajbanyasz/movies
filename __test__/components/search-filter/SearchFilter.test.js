import React from 'react';
import SearchFilter from '../../../src/js/components/search-filter/SearchFilter.jsx';
import { render, mount } from 'enzyme';

describe('SearchFilter', () => {
  it('should rendered correctly', () => {
    const component = render(<SearchFilter searchby={'TITLE'} changeHandler={() => {}} />);
    expect(component).toMatchSnapshot();
  });

  it('should change the component state when selected option changing', () => {
    const component = mount(<SearchFilter searchby={'TITLE'} changeHandler={() => {}} />);
    expect(component.find('.search-by-title').hostNodes().props().className.includes('active')).toBe(true);
    expect(component.state().searchby).toBe(null);
    component.find('.search-by-genre input').hostNodes().simulate('change', {target: {value: 'GENRE'}});
    component.update();
    expect(component.state().searchby).toBe('GENRE');
    component.find('.search-by-title input').hostNodes().simulate('change', {target: {value: 'TITLE'}});
    component.update();
    expect(component.state().searchby).toBe('TITLE');
  });
});