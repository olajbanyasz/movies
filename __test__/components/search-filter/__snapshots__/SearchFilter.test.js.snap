// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter should rendered correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "search-filter",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "SEARCH BY",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "search-filter-buttons btn-toolbar",
            "role": "toolbar",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "btn-group",
                "data-toggle": "buttons",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "search-by-title btn btn-default active",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "TITLE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "TITLE",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "TITLE",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "TITLE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "search-by-genre btn btn-default",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "GENRE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "data": "GENRE",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "text",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      Object {
                        "data": "GENRE",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "autocomplete": "off",
                            "name": "searchfilter",
                            "type": "radio",
                            "value": "GENRE",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "autocomplete": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "autocomplete": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                        },
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "search-by-genre btn btn-default",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "GENRE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "GENRE",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "GENRE",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "GENRE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "search-by-title btn btn-default active",
                      "type": "button",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "TITLE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "data": "TITLE",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "text",
                        },
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      Object {
                        "data": "TITLE",
                        "next": null,
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "autocomplete": "off",
                            "checked": "",
                            "name": "searchfilter",
                            "type": "radio",
                            "value": "TITLE",
                          },
                          "children": Array [],
                          "name": "input",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "autocomplete": undefined,
                            "checked": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "autocomplete": undefined,
                            "checked": undefined,
                            "name": undefined,
                            "type": undefined,
                            "value": undefined,
                          },
                        },
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "type": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-toggle": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-toggle": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      Object {
        "attribs": Object {
          "class": "search-filter-buttons btn-toolbar",
          "role": "toolbar",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "btn-group",
              "data-toggle": "buttons",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "search-by-title btn btn-default active",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "autocomplete": "off",
                      "checked": "",
                      "name": "searchfilter",
                      "type": "radio",
                      "value": "TITLE",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "TITLE",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autocomplete": undefined,
                      "checked": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autocomplete": undefined,
                      "checked": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "data": "TITLE",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "TITLE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "search-by-genre btn btn-default",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "GENRE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "GENRE",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "GENRE",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "GENRE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "search-by-genre btn btn-default",
                  "type": "button",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "autocomplete": "off",
                      "name": "searchfilter",
                      "type": "radio",
                      "value": "GENRE",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "data": "GENRE",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "text",
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "autocomplete": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "autocomplete": undefined,
                      "name": undefined,
                      "type": undefined,
                      "value": undefined,
                    },
                  },
                  Object {
                    "data": "GENRE",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "GENRE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "search-by-title btn btn-default active",
                    "type": "button",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "autocomplete": "off",
                        "checked": "",
                        "name": "searchfilter",
                        "type": "radio",
                        "value": "TITLE",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "data": "TITLE",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "text",
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "autocomplete": undefined,
                        "checked": undefined,
                        "name": undefined,
                        "type": undefined,
                        "value": undefined,
                      },
                    },
                    Object {
                      "data": "TITLE",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "autocomplete": "off",
                          "checked": "",
                          "name": "searchfilter",
                          "type": "radio",
                          "value": "TITLE",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "autocomplete": undefined,
                          "checked": undefined,
                          "name": undefined,
                          "type": undefined,
                          "value": undefined,
                        },
                      },
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "type": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-toggle": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-toggle": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "data": "SEARCH BY",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
